libcontainer/intelrdt/intelrdt.go

```diff
@@ -487,13 +497,21 @@ func (m *Manager) Destroy() error {
 	// Don't remove resctrl group if closid has been explicitly specified. The
 	// group is likely externally managed, i.e. by some other entity than us.
 	// There are probably other containers/tasks sharing the same group.
-	if m.config.IntelRdt != nil && m.config.IntelRdt.ClosID == "" {
-		m.mu.Lock()
-		defer m.mu.Unlock()
-		if err := os.Remove(m.GetPath()); err != nil && !os.IsNotExist(err) {
-			return err
+	if m.config.IntelRdt != nil {
```

Comment:
nit: Let's do a quick return if intelRdt is nil and remove the indentation for the rest of the function